sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -721,6 +721,26 @@ object DataSourceStrategy
           Some(new Sum(FieldReference(name), agg.isDistinct))
         case aggregate.Average(PushableColumnWithoutNestedColumn(name), _) =>
           Some(new GeneralAggregateFunc("AVG", agg.isDistinct, Array(FieldReference(name))))
+        case aggregate.VariancePop(PushableColumnWithoutNestedColumn(name), _) =>
+          Some(new GeneralAggregateFunc("VAR_POP", agg.isDistinct, Array(FieldReference(name))))
+        case aggregate.VarianceSamp(PushableColumnWithoutNestedColumn(name), _) =>
+          Some(new GeneralAggregateFunc("VAR_SAMP", agg.isDistinct, Array(FieldReference(name))))
```

Comment:
FYI: I am adding backticks around `name` in https://github.com/apache/spark/pull/35108. You might want to do the same.

Reply:
Thanks for the remind.

Reply:
Sorry, shouldn't ask you to change yet. I made some changes again https://github.com/apache/spark/pull/35108/commits/49348a95ce3f343aea0b3165a614a5d7db7075d8

Reply:
OK. I reverted.